src/txn/txn_recover.c

```diff
@@ -474,6 +474,13 @@ __recovery_set_checkpoint_snapshot(WT_SESSION_IMPL *session)
     conn = S2C(session);
     counter = 0;
 
+    /*
+     * WiredTiger versions 10.0.1 onward have a valid checkpoint snapshot on-disk. Ignore reading
+     * the on-disk checkpoint snapshot from older versions.
+     */
+    if (conn->recovery_major < 10 || conn->recovery_patch < 1)
```

Comment:
Maybe we need to check minor too for versions like `10.2.0`?

Reply:
For example version - 10.2.0 is greater than 10.0.1, so it is automatically supported. We don't need to check for minor versions separately, because the minor version we started supporting is 0.

Reply:
Sorry, maybe I misunderstand the logic here. If taking `10.2.0` as an example, the expression `conn->recovery_major < 10 || conn->recovery_patch < 1` will evaluate to `true` and we exit from the method. Is that what supposed to happen?

Reply:
Yes. you are correct. Will fix it.

Reply:
Fixed. Please check the new code again.

Reply:
I think `11.0.0` will evaluate `true`. What if we modify the check this way `if (conn->recovery_major < 10 || (conn->recovery_major == 10 && conn->recovery_minor == 0 && conn->recovery_patch == 0))`, do you think it'll work as expected?

Reply:
Yes, it works. Thanks.